### PR TITLE
fix: some fixes for block payee validation and corresponding tests

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
1. we _should not_ skip masternode payments checks below nSuperblockStartBlock or when governance is disabled
2. we _should_ skip superblock payee checks while we aren't synced yet (should help recovering from missed triggers)

## What was done?
pls see individual commits. 

## How Has This Been Tested?
run tests, sync w/ and w/out `--disablegovernance`, reindexed on testnet

## Breaking Changes
n/a

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by ogabrielides — APPROVED** [review]
ACK
Resynced Testet with and without `--disablegovernance`.
Good job 👍

File: src/masternode/payments.cpp [hunk 1]
```diff
@@ -305,14 +310,7 @@ bool IsBlockPayeeValid(const CSporkManager& sporkManager, CGovernanceManager& go
...
     }
 
-    // Check for correct masternode payment
-    if(IsTransactionValid(txNew, pindexPrev, blockSubsidy, feeReward)) {
-        LogPrint(BCLog::MNPAYMENTS, "%s -- Valid masternode payment at height %d: %s", __func__, nBlockHeight, txNew.ToString()); /* Continued */
-        return true;
-    }
-
-    LogPrintf("%s -- ERROR: Invalid masternode payment detected at height %d: %s", __func__, nBlockHeight, txNew.ToString()); /* Continued */
-    return false;
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/masternode/payments.cpp:315]: this `false` moved to https://github.com/dashpay/dash/pull/5684/files/12f3603bd6f381025a5d6415a726421d7f584f8c#diff-fe65493b850e303fea3e506b54fee5bfd4d9238ead17f4cf26afd47ac88447beR274 and invalid superblocks are caught here https://github.com/dashpay/dash/pull/5684/files/12f3603bd6f381025a5d6415a726421d7f584f8c#diff-fe65493b850e303fea3e506b54fee5bfd4d9238ead17f4cf26afd47ac88447beR303

**Review by PastaPastaPasta — COMMENTED** [review]
Local test failures

```
> $ python3 test/functional/feature_governance.py                                                                [±fix_payee_check ●]
2023-11-10T15:31:37.809000Z TestFramework (INFO): Initializing test directory /var/folders/0_/mx_h2k6n0m10356rwgdj1b6h0000gn/T/dash_func_test_b4hi2hpa
2023-11-10T15:31:37.811000Z TestFramework (INFO): Creating and starting controller node
2023-11-10T15:31:38.754000Z TestFramework (INFO): Generating 5100 coins
2023-11-10T15:31:40.813000Z TestFramework (INFO): Creating and starting 0 simple nodes
2023-11-10T15:31:40.814000Z TestFramework (INFO): Activating DIP3
2023-11-10T15:31:40.816000Z TestFramework (INFO): Preparing 5 masternodes
2023-11-10T15:31:41.063000Z TestFramework (INFO): Prepared MN 0: collateral_txid=3ac1880a8db057cb53381956bd779df5c741a6bd1d20b664a4fce9adc220c552, collateral_vout=0, protxHash=613a21b5facfc06e501aefe2e3dd0d44f6d3ee3725717e39970407fd87eaee2c
2023-11-10T15:31:41.433000Z TestFramework (INFO): Prepared MN 1: collateral_txid=ed9d4f2d056690798c03a18a28c22d498d479e105303d1424e52b68507f5e8d9, collateral_vout=1, protxHash=027a33a7a46fef453fc5c609b6c845f4454deace9e25a91e31704039b5eb756d
2023-11-10T15:31:41.803000Z TestFramework (INFO): Prepared MN 2: collateral_txid=06fc4218755dd6be1f1994de3188778cde7cb7520ec3d8c5700064cc132ba3b1, collateral_vout=0, protxHash=f23fb1f236d19c84882e5c48df9ff7119c9b792188b70cd827da4a388852dde3
2023-11-10T15:31:42.172000Z TestFramework (INFO): Prepared MN 3: collateral_txid=d0b5c492d74a69404497df457bf159384f2caad0a8e52efd0d27f553d8e3155d, collateral_vout=1, protxHash=ba6b94a28f5bae9036451950da8be47e691fafbaa00f0c3fad516da1ed053faf
2023-11-10T15:31:42.526000Z TestFramework (INFO): Prepared MN 4: collateral_txid=98f5038274d4ca9993943033cb448a8fa3c1ec2ca15cc88a0f6103bfebc19985, collateral_vout=0, protxHash=dab7cd6406a0620452dbc0b74ec4456cda897e3a4e6c47b32b91741dcf621122
2023-11-10T15:31:44.149000Z TestFramework (INFO): Starting 5 masternodes
2023-11-10T15:31:47.585000Z TestFramework (INFO): Wait for dip0008 activation
2023-11-10T15:34:19.834000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/pasta/workspace/dash/test/functional/test_framework/test_framework.py", line 155, in main
    self.run_test()
  File "test/functional/feature_governance.py", line 315, in run_test
    self.check_superblock()
  File "test/functional/feature_governance.py", line 78, in check_superblock
    assert_equal(payments_found, 2)
  File "/Users/pasta/workspace/dash/test/functional/test_framework/util.py", line 49, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(0 == 2)
2023-11-10T15:34:20.337000Z TestFramework (INFO): Stopping nodes
2023-11-10T15:34:21.381000Z TestFramework (WARNING): Not cleaning up dir /var/folders/0_/mx_h2k6n0m10356rwgdj1b6h0000gn/T/dash_func_test_b4hi2hpa
2023-11-10T15:34:21.381000Z TestFramework (ERROR): Test failed. Test logging available at /var/folders/0_/mx_h2k6n0m10356rwgdj1b6h0000gn/T/dash_func_test_b4hi2hpa/test_framework.log
2023-11-10T15:34:21.381000Z TestFramework (ERROR): 
2023-11-10T15:34:21.382000Z TestFramework (ERROR): Hint: Call /Users/pasta/workspace/dash/test/functional/combine_logs.py '/var/folders/0_/mx_h2k6n0m10356rwgdj1b6h0000gn/T/dash_func_test_b4hi2hpa' to consolidate all logs
2023-11-10T15:34:21.382000Z TestFramework (ERROR): 
2023-11-10T15:34:21.382000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2023-11-10T15:34:21.382000Z TestFramework (ERROR): https://github.com/dashpay/dash/issues
2023-11-10T15:34:21.382000Z TestFramework (ERROR): 
```

**Comment by UdjinM6** [comment]
> Local test failures

Interesting... It fails on develop too sometimes. I'll have a look but imo it's unrelated.

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**